src/librbd/ImageState.cc

```diff
@@ -229,18 +230,191 @@ class ImageUpdateWatchers {
   }
 };
 
+class QuiesceWatchers {
```

Comment:
I think we could move only register/unregister logic in a base class, though right now it is a little different in both classes. It could probably be unified, but it seems we would not gain much with a base class. Still, if you think it is worth a try, I will do.